internal/manifest/oci_index.go

```diff
@@ -76,6 +78,17 @@ func (index *OCI1IndexPublic) UpdateInstances(updates []ListUpdate) error {
 	return index.editInstances(editInstances)
 }
 
+func addCompressionAnnotations(compressionAlgorithm []compression.Algorithm, annotationsMap map[string]string) {
```

Comment:
Hum, should this also remove the annotation if the image is not using zstd? E.g. copying a zstd image while converting to gzip.

(That can only work if call callers correctly fill in the list of algorithms…)

Reply:
“if all callers” …

Reply:
Yes , one hurdle which i saw what defines if user wants to recompress as `gzip` is it leaving the `compressionAlgorithm` list empty or adding a new element `gzip`, Callers would need to take care of this.

I left a `TODO` comment instead of implementing it now.